api/agent/drivers/docker/docker.go

```diff
@@ -48,11 +50,13 @@ func (r *runResult) Error() error   { return r.err }
 func (r *runResult) Status() string { return r.status }
 
 type DockerDriver struct {
-	conf     drivers.Config
-	docker   dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
-	hostname string
-	auths    map[string]docker.AuthConfiguration
-	pool     DockerPool
+	conf         drivers.Config
+	docker       dockerClient // retries on *docker.Client, restricts ad hoc *docker.Client usage / retries
+	hostname     string
+	auths        map[string]docker.AuthConfiguration
+	pool         DockerPool
+	networksLock sync.Mutex
```

Comment:
I assume that this protects the networks map, but it would be great if you would add a comment that states exactly what should be covered by this mutex.

Reply:
done.